python/lsst/daf/butler/formatters/pickleFormatter.py

```diff
@@ -77,4 +77,47 @@ def _writeFile(self, inMemoryDataset, fileDescriptor):
             The file could not be written.
         """
         with open(fileDescriptor.location.path, "wb") as fd:
-            pickle.dump(inMemoryDataset, fd, protocol=-1)
+            fd.write(self._toBytes(inMemoryDataset))
+
+    def _fromBytes(self, inMemoryDataset, pytype=None):
```

Comment:
Argument name is confusing. We are returning an `inMemoryDataset`.

Reply:
Renamed to `bytesObject` is that better? Other suggestions?

Reply:
I still don't see it renamed, does that mean this PR is not up to date? `serializedDataset` is used elsewhere.

Reply:
All these changes came back because I was trying something out and then decided it wasn't a good idea and without thinking did a `git pull` to reset code to origin. Then I remembered that pull will do a merge and quite right the history was very bad. 

I restored a lot of stuff from test branches with `reset --hard` but I guess I forgot to check that formatters were properly up-to-date. That is why so much of this PR seems messy and a step back. I must have missed a commit.